Week3/homework/3-step3.js

```diff
@@ -3,6 +3,15 @@
 // use a 'for' loop
 function repeatStringNumTimesWithFor(str, num) {
   // add your code here
+  if (num < 0) {
+    str = "";
+  } else {
+    let repeatedString = "";
+    for (let i = 0; i < num; i++) {
+      repeatedString += str;
+    }
+    str = repeatedString;
+  }
```

Comment:
Hi @Wael-Alhomsi - a couple of things here. Firstly, I liked the way you wrote the `if-else` block to handle the negative scenarios but you missed a `num === 0` operation. You can maybe add it in the if block itself like `num <= 0`. 

Secondly, as a good developer practice, you should tend to avoid writing to arguments which your function receives. In this case, we should avoid setting `str` anywhere in our function code. As an alternative, you can simply return the `repeatedString`. I saw the freeCodeCamp example and they are returning the same `str`. It works just fine in this case, however, it might cause some issues had `str` been an array or an object. I would maybe suggest playing with the below code snippet. 🙂 
```
const func = arr => {
  arr.push(10);
  return arr;
}

const arr = [7,8,9];
console.log(arr);  // arr before function call
const arr1 = func(arr);
console.log(arr);  // arr after function call
```

Reply:
Hi @yash-kapila ,
Regarding the first part of your review for Week3/homework/3-step3.js, doesn't the 'else' expression includes the num === 0 condition?
As for the second part thank you for that important information (I didn't know it before), but I was only trying to stick to the code they have already provided on freeCode.Camp.